bb_tracking/models.py

```diff
@@ -76,30 +76,56 @@ def to_dict(self):
                     axis_normalizers=(self.axis_normalizers_min_, self.axis_normalizers_max_),
                     diff_normalizer=(self.diff_normalizer_min_, self.diff_normalizer_max_)
                    )
-    
+
     @classmethod
+    # def load(cls, path):
+    #     import joblib
+    #     with open(path, "rb") as f:
+    #         d = joblib.load(f)
+    #     return cls.from_dict(d)
+
     def load(cls, path):
-        import joblib
-        with open(path, "rb") as f:
-            d = joblib.load(f)
-        return cls.from_dict(d)
-    
+        import xgboost as xgb
+
+        # Load the model using XGBoost's native load_model method
+        booster = xgb.XGBClassifier()
+        booster.load_model(path)
+
+        # Convert the booster to a dictionary (if necessary for your class)
+        # Note: This step depends on how your class `cls` is structured and may need adjustment
+        model_dict = booster.__dict__
+
+        # Initialize and return an instance of cls using the model dictionary
+        return cls.from_dict(model_dict)
+
+
     @classmethod
     def from_dict(cls, d):
         classifier = XGBoostRankingClassifier()
         classifier.model = d["model"]
         classifier.axis_normalizers_min_, classifier.axis_normalizers_max_ = d["axis_normalizers"]
         classifier.diff_normalizer_min_, classifier.diff_normalizer_max_ = d["diff_normalizer"]
-        
+
         return classifier
-    
+
+    # def save(self, path):
+    #     import joblib
+    #
+    #     with open(path, "wb") as f:
+    #         joblib.dump(self.to_dict(), f)
+
+
     def save(self, path):
-        import joblib
-        
-        with open(path, "wb") as f:
-            joblib.dump(self.to_dict(), f)
-    
-    
+        # Check if the model is an instance of xgb.Booster or XGBClassifier/XGBRegressor
+        if isinstance(self, xgb.Booster):
+            # Directly save if it's a Booster
+            self.save_model(path)
+        else:
+            # Convert to Booster and save if it's an XGBClassifier/XGBRegressor
+            booster = self.get_booster()
```

Comment:
This should probably be self.model.get_booster(), but it also disregards the axis_normalizers_min_ / max and diff_normalizer_min_ / max which are also parameters of the fit